### DISCUSSION TITLE
Atomi string lights DP ids not matching

### DISCUSSION THREAD

**Discussion by cfiske** [make-all/tuya-local#1432] [Q&A]
Hi,

I have a couple of strings of [Atomi Smart Color String Lights](https://atomismart.com/product/smart-color-string-lights/) I'm trying to add, but they are being detected as the more generic `rgbww_lightbulb` device type. In examining [the Atomi string lights](https://github.com/make-all/tuya-local/blob/main/custom_components/tuya_local/devices/atomi_string_lights.yaml) config I see it has the right set of DPs but the `id`s do not match mine.

The log message I see is:

```
2023-12-20 08:54:44.515 WARNING (MainThread) [custom_components.tuya_local.config_flow] Device matches rgbww_lightbulb with quality of 83%. DPS: {"updated_at": 1703091283.1106973, "1": false, "2": "scene6", "3": 255, "4": 255, "5": "00ff5d008effff", "101": "ffff0502ff4400d500ff"}
```
And when I use the Tuya IoT site to examine the DPs on mine I get:

```
{"1":"Switch","2":"Mode","3":"Brightness","4":"White Temperature","5":"Color Data","101":"颜色"}
```

So it seems like DP ids 20 thru 24 in the config definition are actually 1 thru 5 on my devices.

I could of course edit the config to match mine, but I'd imagine when this support was added it matched someone else's Atomi string lights correctly. Should this then become a new device type?

Thanks!

**Comment by make-all** [comment]
What difference is there between the RGBWW light it is detected as, and this bulb.  If it is just the name, or one additional dp that is not usable, then there is no point adding another config.

But if there is a meaningful difference, then a new config would be the way forward.

**Reply by cfiske** [reply]
So rather than a single bulb, this entity is a string of lights which can do multi-colored patterns/scenes. The `atomi_string_lights.yaml` matches mine exactly and it enumerates all the functions I need, it just does not have the same DP ids. If there is a way to amend that file to say "if it has DP ids 20 thru 24 of these types OR it has ids 1 thru 5 of those same types, then it's a match" it would be ideal. If not, my inclination would be to make something like `atomi_string_lights_rev2.yaml` with identical content but change the DP ids to match mine.
